connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/DataFrameStatFunctions.scala

```diff
@@ -644,25 +644,7 @@ final class DataFrameStatFunctions private[sql] (sparkSession: SparkSession, roo
    * @since 3.5.0
    */
   def bloomFilter(col: Column, expectedNumItems: Long, numBits: Long): BloomFilter = {
-    buildBloomFilter(col, expectedNumItems, numBits, Double.NaN)
-  }
-
-  private def buildBloomFilter(
-      col: Column,
-      expectedNumItems: Long,
-      numBits: Long,
-      fpp: Double): BloomFilter = {
-    def numBitsValue: Long = if (!fpp.isNaN) {
-      BloomFilter.optimalNumOfBits(expectedNumItems, fpp)
-    } else {
-      numBits
-    }
-
-    if (fpp <= 0d || fpp >= 1d) {
```

Comment:
I don't know the reason that remove the check.
It seems is a break change.

Reply:
we don't have such a check in vanilla spark, `BloomFilterAggregate` will check the value range

Reply:
I see. So this change is try to fix the bug.

Reply:
For Vanilla Spark, before the refactoring work https://github.com/apache/spark/pull/43391, there was this check logic when constructing a `BloomFilter` using the following constructor:

https://github.com/apache/spark/blob/bc889c8c2adb976bdf3e7f020e3c7c3de5339c54/common/sketch/src/main/java/org/apache/spark/util/sketch/BloomFilter.java#L244-L251

Reply:
After the refactoring work https://github.com/apache/spark/pull/43391 and https://github.com/apache/spark/pull/44821/files, if an invalid `fpp` value is input, like `df.stat.bloomFilter("id", 1000, -1.0)`

I will see the following error message:

```
[info] org.apache.spark.sql.AnalysisException: 
[DATATYPE_MISMATCH.VALUE_OUT_OF_RANGE] Cannot resolve "bloom_filter_agg(id, 1000, 0)" due to data type mismatch: The numBits must be between [0, positive] (current value = 0L). SQLSTATE: 42K09
```

Personally feel that the new error message is not very user-friendly. My input expression is not `bloom_filter_agg(id, 1000, 0)`, and I did not specify the `numBits` parameter. Why is this error reported? How should I fix it?